aws/resource_aws_gamelift_build.go

```diff
@@ -0,0 +1,186 @@
+	conn := meta.(*AWSClient).gameliftconn
+
+	sl := expandGameliftStorageLocation(d.Get("storage_location").([]interface{}))
+	input := gamelift.CreateBuildInput{
+		Name:            aws.String(d.Get("name").(string)),
+		OperatingSystem: aws.String(d.Get("operating_system").(string)),
+		StorageLocation: sl,
+	}
+	if v, ok := d.GetOk("version"); ok {
+		input.Version = aws.String(v.(string))
+	}
+	log.Printf("[INFO] Creating Gamelift Build: %s", input)
+	var out *gamelift.CreateBuildOutput
+	err := resource.Retry(30*time.Second, func() *resource.RetryError {
+		var err error
+		out, err = conn.CreateBuild(&input)
+		if err != nil {
+			if isAWSErr(err, gamelift.ErrCodeInvalidRequestException, "Provided build is not accessible.") {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*out.Build.BuildId)
+
+	stateConf := resource.StateChangeConf{
+		Pending: []string{gamelift.BuildStatusInitialized},
+		Target:  []string{gamelift.BuildStatusReady},
+		Timeout: 1 * time.Minute,
+		Refresh: func() (interface{}, string, error) {
+			out, err := conn.DescribeBuild(&gamelift.DescribeBuildInput{
+				BuildId: aws.String(d.Id()),
+			})
+			if err != nil {
+				return 42, "", err
+			}
+
+			return out, *out.Build.Status, nil
+		},
+	}
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsGameliftBuildRead(d, meta)
+}
+
+func resourceAwsGameliftBuildRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).gameliftconn
+
+	log.Printf("[INFO] Reading Gamelift Build: %s", d.Id())
+	out, err := conn.DescribeBuild(&gamelift.DescribeBuildInput{
+		BuildId: aws.String(d.Id()),
+	})
+	if err != nil {
+		if isAWSErr(err, gamelift.ErrCodeNotFoundException, "") {
+			log.Printf("[WARN] Gamelift Build (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+	b := out.Build
+
+	d.Set("name", b.Name)
+	d.Set("operating_system", b.OperatingSystem)
+	d.Set("version", b.Version)
+
+	return nil
+}
+
+func resourceAwsGameliftBuildUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).gameliftconn
+
+	log.Printf("[INFO] Updating Gamelift Build: %s", d.Id())
+	input := gamelift.UpdateBuildInput{
+		BuildId: aws.String(d.Id()),
+		Name:    aws.String(d.Get("name").(string)),
+	}
+	if v, ok := d.GetOk("version"); ok {
+		input.Version = aws.String(v.(string))
+	}
+
+	_, err := conn.UpdateBuild(&input)
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsGameliftBuildRead(d, meta)
+}
+
+func resourceAwsGameliftBuildDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).gameliftconn
+
+	log.Printf("[INFO] Deleting Gamelift Build: %s", d.Id())
+	_, err := conn.DeleteBuild(&gamelift.DeleteBuildInput{
+		BuildId: aws.String(d.Id()),
+	})
+	if err != nil {
```

Comment:
Nitpick: We should ignore missing build IDs:
```
if err != nil {
  if isAwsErr(err, gamelift.ErrCodeNotFoundException, "") {
    return nil
  }
  return err
}
```

Reply:
We could, but `DeleteBuild` doesn't ever return such error if you try deleting build which doesn't exist. 🤷‍♂️ 

It just returns `200 OK`.